src/components/molecules/email-form.tsx

```diff
@@ -12,33 +13,63 @@ import {
 import { Popover, PopoverContent, PopoverTrigger } from '@/components/atoms/popover.tsx'
 import { RichTextEditor } from '@/components/atoms/rich-text-editor.tsx'
 import { Typography } from '@/components/atoms/typography.tsx'
-import { Vendor } from '@/schemas/vendor.ts'
+import { useToast } from '@/hooks/use-toast.ts'
+import { useBlastEmailMutation } from '@/lib/redux/features/vendor/api.ts'
+import { EmailVendorsArgs } from '@/lib/redux/features/vendor/validation.ts'
+import { toastForError } from '@/lib/redux/utils.tsx'
+
+import { Input } from '../atoms/input.tsx'
+import { Label } from '../atoms/label.tsx'
 
 interface EmailFormProps {
-  selectedVendors: Vendor[]
-  onClose: () => void
+  vendorIds: string[]
   defaultContent?: string
+  toggleDialog: boolean
+  setToggleDialog: Dispatch<SetStateAction<boolean>>
 }
 
 export function EmailForm({
-  selectedVendors,
-  onClose,
+  vendorIds,
   defaultContent = '',
+  toggleDialog,
+  setToggleDialog,
 }: Readonly<EmailFormProps>) {
-  const [isConfirmation, setIsConfirmation] = useState(false)
-  const [emailContent, setEmailContent] = useState(defaultContent)
+  const [isConfirmation, setIsConfirmation] = useState<boolean>(false)
+  const [emailBody, setEmailBody] = useState<string>(defaultContent)
+  const [emailSubject, setEmailSubject] = useState<string>('')
   const [showPopover, setShowPopover] = useState(false)
 
-  const handleSendEmail = () => { // TODO: Replace the simulated email sending logic with a real implementation
-    selectedVendors.forEach((vendor) => {
-      console.log(`Sending email to ${vendor.name}:`, emailContent)
-    })
-
-    onClose()
+  const [blastEmail] = useBlastEmailMutation()
+  const { toast } = useToast()
+  const handleEmailVendors = async () => {
+    try {
+      const args: EmailVendorsArgs = {
+        vendor_ids: vendorIds,
+        email_template: {
+          body: emailBody,
+          subject: emailSubject,
+        },
+      }
+      toast({
+        title: 'On Progress',
+        description: 'Executing email blast',
+        duration: 2000,
+      })
+      await blastEmail(args).unwrap()
+      toast({
+        title: 'Success',
+        description: 'Email blast has successfully executed',
+        duration: 2000,
+      })
+      setToggleDialog(false)
+    }
+    catch (_) {
+      toastForError(new Error('Email blast failed to be executed'))
+    }
```

Comment:
This is not how we use toastForError, the correct way is to pass the exception caught by the catch block and pass it to the `toastForError` function

```ts
catch (err) {
  toastForError(err)
}
```
the function handles all sort of rtk query error logic